circuits/cpp/src/aztec3/circuits/kernel/private/testing_harness.cpp

```diff
@@ -42,6 +39,69 @@ using aztec3::circuits::abis::TxContext;
 using aztec3::circuits::abis::TxRequest;
 using aztec3::circuits::abis::private_kernel::PrivateCallData;
 
+using aztec3::utils::array_length;
+using aztec3::utils::zero_array;
+
+/**
+ * @brief Get the random read requests and their membership requests
+ *
+ * @details read requests are siloed by contract address before being
+ * inserted into mock private data tree
+ *
+ * @param contract_address address to use when siloing read requests
+ * @param num_read_requests if negative, use random num. Must be < READ_REQUESTS_LENGTH
+ * @return std::tuple<read_requests, read_request_memberships_witnesses, historic_private_data_tree_root>
+ */
+std::tuple<std::array<NT::fr, READ_REQUESTS_LENGTH>,
+           std::array<MembershipWitness<NT, PRIVATE_DATA_TREE_HEIGHT>, READ_REQUESTS_LENGTH>,
+           NT::fr>
+get_random_reads(NT::fr const& contract_address, int const num_read_requests)
+{
+    auto read_requests = zero_array<fr, READ_REQUESTS_LENGTH>();
+    auto leaves = zero_array<fr, READ_REQUESTS_LENGTH>();
+    // randomize the number of read requests with a configurable minimum
+    auto final_num_rr = num_read_requests >= 0
```

Comment:
This has a configurable minimum but the return type is fixed to length 4, should there be a ceiling? I guess this is a test so it doesnt matter but may be a footgun.